flutter_theoplayer_sdk_android/lib/theoplayer_android.dart

```diff
@@ -15,57 +16,84 @@ class THEOplayerAndroid extends TheoplayerPlatform {
   }
 
   @override
-  Widget buildView(BuildContext context, THEOplayerConfig theoPlayerConfig, THEOplayerViewCreatedCallback createdCallback) {
+  void initalize(THEOplayerConfig theoPlayerConfig, InitializeNativeResultCallback callback) {
+    if (!theoPlayerConfig.androidConfig.viewComposition.isPlatformView) {
+      // Pass parameters to the platform side.
+      Map<String, dynamic> creationParams = <String, dynamic>{};
+      creationParams["playerConfig"] = theoPlayerConfig.toJson();
+
+      PlatformPlayersService.instance.createPlayer(creationParams).then((value) =>
+          callback(value)
+      );
+    } else {
+      // return -1 as default, as the [buildView] method doesn't care about this anymore for PlatformViews
+      callback(-1);
+    }
+  }
+
+  @override
+  Widget buildView(BuildContext context, THEOplayerConfig theoPlayerConfig, THEOplayerViewCreatedCallback createdCallback, int textureId) {
     // This is used in the platform side to register the view.
     const String viewType = 'com.theoplayer/theoplayer-view-native';
 
     // Pass parameters to the platform side.
     Map<String, dynamic> creationParams = <String, dynamic>{};
     creationParams["playerConfig"] = theoPlayerConfig.toJson();
 
-    return PlatformViewLink(
-      viewType: viewType,
-      surfaceFactory: (context, controller) {
-        return AndroidViewSurface(
-          controller: controller as AndroidViewController,
-          gestureRecognizers: const <Factory<OneSequenceGestureRecognizer>>{},
-          hitTestBehavior: PlatformViewHitTestBehavior.opaque,
-        );
-      },
-      onCreatePlatformView: (params) {
-        late AndroidViewController androidViewController;
-
-        if (theoPlayerConfig.androidConfig.useHybridComposition) {
-          androidViewController = PlatformViewsService.initExpensiveAndroidView(
-            id: params.id,
-            viewType: viewType,
-            layoutDirection: TextDirection.ltr,
-            creationParams: creationParams,
-            creationParamsCodec: const StandardMessageCodec(),
-            onFocus: () {
-              params.onFocusChanged(true);
-            },
-          );
-        } else {
-          androidViewController = PlatformViewsService.initAndroidView(
-            id: params.id,
-            viewType: viewType,
-            layoutDirection: TextDirection.ltr,
-            creationParams: creationParams,
-            creationParamsCodec: const StandardMessageCodec(),
-            onFocus: () {
-              params.onFocusChanged(true);
-            },
+    if (!theoPlayerConfig.androidConfig.viewComposition.isPlatformView) {
+      var texture = Texture(textureId: textureId);
+      Future.delayed(Duration.zero, (){ // wait for the next tick, TODO: check microTasks
+        var controller = THEOplayerViewControllerAndroid(textureId);
+        // trigger the surface attachment
+        controller.configureSurface(textureId, 0, 0);
```

Comment:
maybe it's a bit weird here with 0,0, but the Texture figured out the size already by itself and takes up all the available space.